tests/std/syntax/printer.test.luau

```diff
@@ -0,0 +1,236 @@
+local luau = require("@std/luau")
+local parser = require("@std/syntax/parser")
+local printer = require("@std/syntax/printer")
+local query = require("@std/syntax/query")
+local syntaxTypes = require("@std/syntax/types")
+local syntaxUtils = require("@std/syntax/utils")
+local test = require("@std/test")
+
+test.suite("syntax_printer", function(suite)
+	suite:case("string_replacement", function(assert)
+		local source = [[
+            local name = "World"
+        ]]
+
+		local ast = parser.parse(source)
+		local replacements = query.select(ast, syntaxUtils.isExprConstantString):replace(function(s)
+			return if s.text == "World" then "1 + 2" else nil
+		end)
+
+		local result = printer.printstatement(ast, replacements)
+
+		local expected = "            local name = 1 + 2" -- TODO: preserve trivia so trailing newline is preserved
```

Comment:
That's not a `TODO`, that's an important part of the contract for dealing with CSTs. Also what happened to the newline at the beginning of the string?